vulnerabilities/importers/elixir_security.py

```diff
@@ -0,0 +1,120 @@
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+
+import yaml
+import re
+import json
+import requests
+from typing import Set
+from typing import List
+from dephell_specifier import RangeSpecifier
+from packageurl import PackageURL
+
+from vulnerabilities.data_source import GitDataSource
+from vulnerabilities.data_source import GitDataSourceConfiguration
+from vulnerabilities.data_source import Advisory
+from vulnerabilities.data_source import Reference
+
+
+class ElixirSecurityDataSource(GitDataSource):
+    def __enter__(self):
+        super(ElixirSecurityDataSource, self).__enter__()
+
+        if not getattr(self, "_added_files", None):
+            self._added_files, self._updated_files = self.file_changes(
+                recursive=True, file_ext="yml", subdir="./packages"
+            )
+
+    def updated_advisories(self) -> Set[Advisory]:
+        files = self._updated_files
+        advisories = []
+        for f in files:
+            processed_data = self.process_file(f)
+            if processed_data:
+                advisories.append(processed_data)
+        return self.batch_advisories(advisories)
+
+    def added_advisories(self) -> Set[Advisory]:
+        files = self._added_files
+        advisories = []
+        for f in files:
+            processed_data = self.process_file(f)
+            if processed_data:
+                advisories.append(processed_data)
+        return self.batch_advisories(advisories)
+
+    @staticmethod
+    def generate_all_version_list(pkg_name):
+        resp = requests.get(f"https://hex.pm/api/packages/{pkg_name}")
+        resp = resp.content
+        json_resp = json.loads(resp)
+        version_list = []
+        for release in json_resp["releases"]:
+            version_list.append(release["version"])
+        return version_list
+
+    def get_pkg_from_range(self, version_list, pkg_name):
+        pkg_versions = []
+        all_version_list = self.generate_all_version_list(pkg_name)
+        if version_list is None:
```

Comment:
How does this even get satisfied ?

Reply:
There's a line below which says 

```
safe_pkg_versions = self.get_pkg_from_range(
                    yaml_file["patched_versions"] + yaml_file["unaffected_versions"],
                    pkg_name,
                )
```
IMHO when you add None  to anything that should cause exception there. Why check for None again ?

Reply:
Ah I see  that's probably for the line below that.

Reply:
That is if the version_list passed to the function is none